### ISSUE TITLE
Improve unit tests for Ops

### ISSUE THREAD

**Issue by sgligorijevicTT** [tenstorrent/tt-mlir#1498]
Our unit tests are inconsistent and could be improved. The simplest tests merely check for the existence of empty op and the expected op, with no other checks. Slightly better tests have hardcoded shapes and types.
Ideally, all tests would capture the types of arguments(and possibly the return type) from the func declaration and verify that all ops from the output have the right types in the right places. It is further possible to capture individual axis sizes to allow us to check that shapes relate to each other in expected ways. This is useful for operations like matmul for example.

**Comment by skhatriTT** [comment]
@sgligorijevicTT has this been worked on?

**Comment by sgligorijevicTT** [comment]
No. Improving tests is still relevant, just a very low priority.
